### ISSUE TITLE
wp-secrets.php is not persisted and only generated when the volume is empty

### ISSUE THREAD

**Issue by bb** [TrafeX/docker-wordpress#42]
Hi @TrafeX, thanks for this great image. After a first investigation, it looks muuuuch better than the official ones.

When migrating a few sites from https://hub.docker.com/_/wordpress to this image, I noticed an issue with the secrets which is already outlined in #36. As this was closed long time ago and didn't receive any update, I decided to open this here.

So, the persistent volume is mounted to `/var/www/wp-content`, but the secrets are in `/usr/src/wordpress/wp-secrets.php` which is NOT part of the volume. When updating or recreating the container, the file reverts to the original state without secrets defined:

```sh
docker-compose exec wordpress bash -l
curl -f https://api.wordpress.org/secret-key/1.1/salt/ >> /usr/src/wordpress/wp-secrets.php
exit
docker-compose up -d --force-recreate wordpress
docker-compose exec wordpress bash -l
cat /usr/src/wordpress/wp-secrets.php
```

Output of last command shows:
```php
<?php
/** Generated file based on https://api.wordpress.org/secret-key/1.1/salt/ */
```

This is obviously missing the secrets. Besides `--force-recreate` this also happens when pulling a new image version or changing an environment variable for the container.

Same issue happens with a copied-over volume as described in https://github.com/TrafeX/docker-wordpress/issues/36#issuecomment-1570325428 -- however in contrast to @mdbraber I think secrets should NOT be regenerated during image updates but persisted and manually rotated.

I now considered setting the constants from https://api.wordpress.org/secret-key/1.1/salt/ as environment variables because they are currently passed to `define` in wp-config.php. However, if this ticket is fixed, this would lead to warnings, so maybe this combination is something to keep in mind when fixing it.

**Comment by Morpheus636** [comment]
I think setting it with environment variables would be a really good option. Preventing errors would be relatively simple, either via a conditional on the import in wp-config.php or on the actual generation in entrypoint.sh.

**Comment by TrafeX** [comment]
@bb This should now be fixed when you pull in the `latest` tag. Would be nice if you can give it a try, I'll tag it in a release after some more testing.

**Comment by bb** [comment]
@TrafeX thanks for following up on this. I reviewed the PR #50 and like your approach with persisting+symlinking the file and regenerating whenever the file does not exist.

I'm not completely happy with the approach of checking if _any_ of the variables exist, though. While this is way better than before, I'd say it should be all-or-none of the environenment variables. If e.g. $AUTH_KEY is defined but $SECURE_AUTH_KEY is not, it should e.g. not start up at all but print an error message. But currently it will start up and not generate the wp-secrets file.

However, personally I'm not using this anymore, so I don't really care. I moved on in a similar direction like  @henrygd's fork and the official wordpress images, i.e. I'm not only persisting wp-content but the whole wordpress installation.
